Server/htdocs/AppController/commands_RSM/utilities/RSMitemsManagement.php

```diff
@@ -3495,4 +3495,59 @@ function recalculateOrder($oldValue, $newValue, $oldOrder) {
 
     return implode(',', $newOrders);
 }
+
+// Returns true if the item exists 
+function verifyItemExists($ID, $itemTypeID, $clientID) {
+    $result = RSQuery("SELECT RS_ITEM_ID FROM rs_items WHERE RS_ITEMTYPE_ID = " . $itemTypeID . " AND RS_ITEM_ID = " . $ID . " AND RS_CLIENT_ID = " . $clientID);
+    return !empty($result->fetch_assoc());
+}
+
+function getPropertiesExtendedForToken($itemTypeID, $RStoken, $propertyIDs) {
+
+    // build in clause for propertyids
+    $formattedPropertyIDs = '(';
+    foreach ($propertyIDs as $prop) {
+        $formattedPropertyIDs .= $prop['ID'] . ',';
+    }
+    $formattedPropertyIDs = rtrim($formattedPropertyIDs, ',') . ')';
+
+    $clientID = RSclientFromToken($RStoken);
+
+    // build a fast query to get user properties
+    $theQuery_getProperties = 'SELECT DISTINCT rs_categories.RS_NAME AS "cName",
+                                         rs_categories.RS_ORDER,
+                                         rs_item_properties.RS_PROPERTY_ID AS "pID",
+                                         rs_item_properties.RS_ORDER
+        FROM rs_categories
+                INNER JOIN rs_item_properties USING (RS_CLIENT_ID, RS_CATEGORY_ID)
+                INNER JOIN rs_token_permissions USING (RS_CLIENT_ID, RS_PROPERTY_ID)
+                INNER JOIN rs_tokens USING (RS_CLIENT_ID)
+        WHERE rs_categories.RS_ITEMTYPE_ID                 =     ' . $itemTypeID . '
+                AND rs_categories.RS_CLIENT_ID                 =     ' . $clientID . '
+                AND rs_item_properties.RS_CLIENT_ID        =     ' . $clientID . '
+                AND rs_token_permissions.RS_CLIENT_ID    =     ' . $clientID . '
+                AND rs_tokens.RS_TOKEN                                 = \'' . $RStoken . '\'
+                AND rs_item_properties.RS_PROPERTY_ID IN ' . $formattedPropertyIDs . '
+
+        ORDER BY rs_categories.RS_ORDER, rs_item_properties.RS_ORDER';
+
+    // execute query
+    $theProperties = RSquery($theQuery_getProperties);
+    // get properties values
+    $results = array();
+
+    $categoryName = '';
+    if ($theProperties) {
+        while ($row = $theProperties->fetch_assoc()) {
+            // Only return properties that user have READ permission
+            if (RShasTokenPermission($RStoken, $row['pID'], "READ")) {
```

Comment:
Esta llamada la evitamos añadiendo AND rs_token_permissions.RS_PERMISSION ='READ' en la query anterior